test/ros_sec_test/runner/test_runner.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+
+#include "runner/runner.hpp"
+#include "test_utilities/utility_fixtures.hpp"
+#include "test_utilities/periodic_attack_component_mock.hpp"
+
+using LifecycleNodeShPtr = std::shared_ptr<rclcpp_lifecycle::LifecycleNode>;
+using ros_sec_test::test::test_utilities::ROSTestingFixture;
+using ros_sec_test::runner::Runner;
+using CallbackReturn = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn;
+using ::testing::Return;
+using ::testing::_;
+
+TEST_F(ROSTestingFixture, test_multiple_attack_lifecycle) {
+  std::vector<LifecycleNodeShPtr> nodes;
+  auto node1 = std::make_shared<MockPeriodicAttackComponent>("first_attack_node");
+  auto node2 = std::make_shared<MockPeriodicAttackComponent>("second_attack_node");
+  nodes.push_back(node1);
```

Comment:
Nit: you could also write

```
auto node1 = std::make_shared<MockPeriodicAttackComponent>("first_attack_node");
auto node2 = std::make_shared<MockPeriodicAttackComponent>("second_attack_node");
auto nodes = {node1, node2};
Runner runner(nodes);
```

Reply:
Ok, I thought it would convert implicitly but seems like it's not working so we can keep it as it is.

Reply:
Note: another solution would be to update the constructor to take a `std::initializer_list<rclcpp_lifecycle::LifecycleNode::SharedPtr>`.

Reply:
re-opening in case you want to try the initializer list route :)